dev-util/ragel/ragel-7.0.0.10.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="Compiles executable finite state machines from regular languages"
+DESCRIPTION="Ragel compiles executable finite state machines from regular \
+languages. Ragel targets C, C++, Objective-C, C#, D, Java, Ruby, OCaml and Go.
+Ragel state machines can not only recognize byte sequences as regular \
+expression machines do, but can also execute code at arbitrary points in the \
+recognition of a regular language.
+Code embedding is done using inline operators that do not disrupt the regular \
+language syntax."
+HOMEPAGE="https://www.colm.net/open-source/ragel/"
+COPYRIGHT="2001-2016 Adrian Thurston
+	2007 Victor Hugo Borja
+	2011 Josef Goettgens"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://www.colm.net/files/ragel/ragel-$portVersion.tar.gz"
+CHECKSUM_SHA256="40562bcac66a22dbea8357a35745bbcb1ab596c262d8691145ee11aafa6f8dec"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	commandSuffix=$secondaryArchSuffix
+	commandBinDir=$binDir
+else
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
```

Comment:
Suggested variation

```sh
commandSuffix=$secondaryArchSuffix
commandBinDir=$binDir
if [ -z "${ARCHITECTURES##*!$targetArchitecture*}" ]; then
	commandSuffix=
	commandBinDir=$prefix/bin
fi

Reply:
What about this variation?
```
commandBinDir=$binDir
commandSuffix=$secondaryArchSuffix
if [ "$targetArchitecture" = x86_gcc2 ]
then
	commandBinDir=$prefix/bin
	commandSuffix=
fi
```
Remark: the commandBinDir "trick" should **_only_** be used **_if_** `ARCHITECTURES` contains `!x86_gcc2`.

Reply:
> Remark: the commandBinDir "trick" should **_only_** be used **_if_** `ARCHITECTURES` contains `!x86_gcc2`.

Hmm, the truth is that we **_could_** use it **_even_** if  `ARCHITECTURES` contains `x86_gcc2` or `?x86_gcc2`, but we would need to add **_this_** after the `REQUIRES`:
```
if [ -n "$secondaryArchSuffix" ]; then
CONFLICTS="
	ragel
	"
fi
```
Important: In the above snippet it's really `ragel` that we need to have in `CONFLICTS`. Because putting `ragel$secondaryArchSuffix` would make ragel_x86 conflict with itself ;-)
(The purpose is to make the secondary arch package conflict with the primary arch package.)

Reply:
Well, that is getting into nonsense territory. The reason we introduced the $secondaryArchSuffix is precisely to not have this kind of conflicts and allow packages to be installed side by side. Why would one insist so hard on removing that feature?

Reply:
I'm not saying we should do so. I'm just saying that, in theory, we could do that. But I do not recommend using the commandBinDir "trick" for recipes which are not broken on a primary arch.